src/librender/imageblock.cpp

```diff
@@ -168,6 +168,25 @@ ImageBlock<Float, Spectrum>::put(const Point2f &pos_, const Float *value, Mask a
     return active;
 }
 
+MTS_VARIANT void
+ImageBlock<Float, Spectrum>::overwrite_channel(size_t channel,
+                                               ScalarFloat value) {
+    size_t pixel_count = (width() + m_border_size) * (height() + m_border_size);
+    Assert(m_data.size() == m_channel_count * pixel_count);
+
+    // TODO: any way to avoid this specialization?
```

Comment:
`m_data` being an `DynamicBuffer<Float>` you should be able to use `ek::scatter` for the `scalar_*` variants as well. Maybe something like
```c++
auto indices = m_channel_count * ek::arange<DynamicBuffer<UInt64>>(pixel_count) + channel;
ek::scatter(m_data, value, indices);
```

Reply:
Thanks for the tip, I just had to add a cast because `value` was a scalar:

```
    auto indices =
        m_channel_count * ek::arange<DynamicBuffer<UInt64>>(pixel_count) + channel;
    ek::scatter(m_data, DynamicBuffer<Float>(value), indices);
```